### ISSUE TITLE
function Bureaucrat.SwaggerSlateMarkdownWriter.write/2 is undefined

### ISSUE THREAD

**Issue by CallMeSH** [api-hogs/bureaucrat#20]
Hello everyone!

I'm trying Bureaucrat right now and I'm facing an issue on a new Phoenix app.
I inserted bureaucrat as a dependency in my mix.exs file.
I added in the test helper the right lignes to make Bureaucrate start and find my already existing swagger file.
Finally as a test I added a `doc` call to one of my test.

Unfortunately, when running them, this error occurs:
`** (UndefinedFunctionError) function Bureaucrat.SwaggerSlateMarkdownWriter.write/2 is undefined (module Bureaucrat.SwaggerSlateMarkdownWriter is not available)`

Do you have any ideas how I can fix this? It's like if this module was not included in the project...
Thanks!

**Comment by mbuhot** [comment]
@CallMeSH It looks like there hasn't been an official release to Hex.pm since April 2016.
The swagger support was added in June, so my guess is that you will need to change your dependency to use github:

```elixir
def deps do
[
  ... usual phoenix dependencies ... 
  {:bureaucrat,  github: "api-hogs/bureaucrat", only: :test},
]
```

@OpakAlex / @arnodirlam Is it time for a new release given the recent enhancements for phoenix channels ?

**Comment by OpakAlex** [comment]
@mbuhot @CallMeSH i will do it on this weekends, will be version 0.2.0.

**Comment by CallMeSH** [comment]
Thanks for the info!
Using the master branch I still face an issue on doc generation:
```
12:06:20.608 [error] GenEvent handler Bureaucrat.Formatter installed in #PID<0.301.0> terminating
** (Protocol.UndefinedError) protocol Enumerable not implemented for nil
    (elixir) lib/enum.ex:1: Enumerable.impl_for!/1
    (elixir) lib/enum.ex:116: Enumerable.reduce/3
    (elixir) lib/enum.ex:868: Enum.find/3
    (bureaucrat) lib/bureaucrat/swagger_slate_markdown_writer.ex:220: Bureaucrat.SwaggerSlateMarkdownWriter.write_operations_for_tag/4
    (elixir) lib/enum.ex:645: Enum."-each/2-lists^foreach/1-0-"/2
    (elixir) lib/enum.ex:645: Enum.each/2
    (elixir) lib/enum.ex:1233: anonymous fn/3 in Enum.map/2
    (stdlib) lists.erl:1263: :lists.foldl/3
    (elixir) lib/enum.ex:1772: Enum.map/2
    (bureaucrat) lib/bureaucrat/formatter.ex:10: Bureaucrat.Formatter.handle_event/2
    (stdlib) gen_event.erl:533: :gen_event.server_update/4
    (stdlib) gen_event.erl:515: :gen_event.server_notify/4
    (stdlib) gen_event.erl:256: :gen_event.handle_msg/5
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {:suite_finished, 462624, nil}
```

It seems like the swagger.json parsing is failing somehow.
The swagger.json is generated threw phoenix_swagger:
```
{
  "swagger": "2.0",
  "paths": {
    "/api/places": {
      "get": {
        "tags": [
          "Places"
        ],
        "summary": "Query for places",
        "responses": {
          "200": {
            "description": "OK"
          }
        },
        "produces": [
          "application/json"
        ],
        "parameters": [],
        "operationId": "list_places",
        "description": "Get the list of all places"
      }
    }
  },
  "info": {
    "version": "0.1",
    "title": "Test"
  },
  "host": "localhost:4000",
  "definitions": {}
}
```

```
 conn = conn
    |> get(place_path(conn, :index))
    |> doc(operation_id: "list_places")
```

**Comment by mbuhot** [comment]
@CallMeSH, I think this code is the issue:

```elixir
  def write_operations_for_tag(file, tag, records_by_operation_id, swagger) do
    tag_details = swagger["tags"] |> Enum.find(&(&1["name"] == tag))

    file
    |> puts("# #{tag}\n")
    |> puts("#{tag_details["description"]}\n")
```

It is assuming that there is a `tags` section in the swagger document.
The quick fix is to add something to your swagger document like [the example](https://github.com/api-hogs/bureaucrat/blob/master/examples/swagger_demo/web/router.ex#L19-L21)

```
def swagger_info do
    %{
      info: %{
        version: "1.0",
        title: "SwaggerDemo App"
      },
      tags: [
        %{name: "User", description: "Operations related to Users"}
      ]
    }
  end
```

Even better if you can submit a PR to handle this cases in the code where `swagger["tags"]` is `nil` or there is no item in the tags list matching the given tag :D

**Comment by CallMeSH** [comment]
Hey! Well there are a few things I would like to change in the framework so I'm probably going to PR soon. Meanwhile, thanks for the help!

**Comment by OpakAlex** [comment]
@CallMeSH published 0.2.0
